infrastructure/restapi/src/main/java/tech/pegasys/teku/infrastructure/restapi/endpoints/EndpointMetadata.java

```diff
@@ -484,6 +505,14 @@ public EndpointMetaDataBuilder queryListParam(final ParameterMetadata<?> paramet
       return this;
     }
 
+    public EndpointMetaDataBuilder header(final ParameterMetadata<?> headerMetadata) {
```

Comment:
Some Beacon API methods have optional headers. We should be able to define a header as optional in the metadata.

Reply:
Regardless if the header is required or not, we probably don't want to be able to specify multiple headers with the same name. So we need to check on both.